awscli/customizations/eks/update_kubeconfig.py

```diff
@@ -287,17 +287,22 @@ def get_user_entry(self):
         Return a user entry generated using
         the previously obtained description.
         """
+
+        region = self._get_cluster_description().get("arn").split(":")[3]
+
         generated_user = OrderedDict([
             ("name", self._get_cluster_description().get("arn", "")),
             ("user", OrderedDict([
                 ("exec", OrderedDict([
                     ("apiVersion", API_VERSION),
                     ("args",
                         [
+                            "--region",
+                            region,
```

Comment:
When the region is set to a non-`us-east-1` region such as `us-west-2`, what is the endpoint url in the presigned url? If it is signed for the global endpoint (i.e.`sts.amazonaws.com`), we should be fine. But if it is using the regional endpoint (i.e. `sts.us-west-2.amazonaws.com`), then there may be some backwards compatiblity concerns if the account does not have the regional endpoints activated. I'm also assuming the previous executable`aws-iam-authenticator` only ever signed for the global STS endpoint.

Reply:
For non-us-east-1, [`get_token`](https://github.com/aws/aws-cli/blob/develop/awscli/customizations/eks/get_token.py#L95-L128) already uses the regional endpoint. This change is to match the behavior from `aws-iam-authenticator` where the customer might not have the AWS cli configured with a region

Reply:
I just double checked this. We should be fine here. Looking at the url in the base64 encoded from running a command such as:
```
aws eks get-token --cluster-name foo --region us-west-2
```
The URL that gets presigned uses `https://sts.amazonaws.com/` and signs with `us-east-1`, which is what I would expect for backwards compatibility with `aws-iam-authenticator`.